libiocage/lib/Distribution.py

```diff
@@ -97,6 +101,41 @@ def _filter_available_releases(self, release_name):
         arch = release_name.split("-")[-2:][0]
         return self.host.processor == arch
 
+    def _get_eol_list(self, releases: List[str]) -> List[str]:
+        """Scrapes the FreeBSD website and returns a list of EOL RELEASES"""
+        _eol = "https://www.freebsd.org/security/unsupported.html"
+        req = requests.get(_eol)
+        status = req.status_code == requests.codes.ok
+        eol_releases = []
+        if not status:
+            req.raise_for_status()
+
+        for eol in req.content.decode("iso-8859-1").split():
+            eol = eol.strip("href=").strip("/").split(">")
+            # We want a dynamic EOL
+            try:
+                if "-RELEASE" in eol[1]:
+                    eol = eol[1].strip('</td')
+                    if eol not in eol_releases:
+                        eol_releases.append(eol)
+            except IndexError:
+                pass
+
+        return self._get_eol_releases(releases, eol_releases)
+
+    def _get_eol_releases(self, releases: List[str], eol: List[str]) -> \
+            List[str]:
+        releases_eol = []
+        if self.host.distribution.name == "FreeBSD":
+            for rel in releases:
+                if rel in eol:
+                    rel = f"{rel} (EOL)"
```

Comment:
I suggest to add a property to Release that is set to True when the Release is EOL. That way we don't need to change the internal name. The Release name is involved in calculating the dataset location and much more.

When rendering you can just check

```python
release_name = release.name
if release.eol is True:
  printed_name += " (EOL)"
```

Or even another property on Release that gives you a releases name with annotations for printing:

```
@property
def annotated_name(self):
  annotations = set()
  if self.eol is True:
    annotations.add("EOL")
  if len(annotations) > 0:
    return "{} ({})".format(release.name, ", ".join(annotations))
  return f"{self.name} {suffix}"
```

By using the latter, we could make sure `(EOL)` gets even printed when listing existing jails, if we want to.

Reply:
Great suggestion!